.github/workflows/build.yml

```diff
@@ -0,0 +1,321 @@
+
+    steps:
+    - name: Install external dependencies with yum
+      run: |
+        yum update -y
+        yum install --nogpg -y \
+            cmake3 \
+            make \
+            krb5-devel \
+            libuuid-devel \
+            libxml2-devel \
+            openssl-devel \
+            systemd-devel \
+            zlib-devel \
+            devtoolset-7-gcc-c++ \
+            which \
+            python3-devel \
+            python3-setuptools \
+            git \
+            cppunit-devel
+        yum clean all
+
+    # Need to use v1 of action as image Git is too old
+    - name: Clone repository now that Git is available
+      uses: actions/checkout@v1
+
+    - name: Build with cmake
+      run: |
+        . /opt/rh/devtoolset-7/enable
+        cd ..
+        cmake3 \
+            -DCMAKE_INSTALL_PREFIX=/usr/local/ \
+            -DPYTHON_EXECUTABLE=$(command -v python3) \
+            -DENABLE_TESTS=ON \
+            -S xrootd \
+            -B build
+        cmake3 build -LH
+        cmake3 \
+            --build build \
+            --clean-first \
+            --parallel $(($(nproc) - 1))
+        cmake3 --build build --target install
+        python3 -m pip list
+
+    - name: Verify install
+      run: |
+        command -v xrootd
+        command -v xrdcp
+
+    - name: Verify Python bindings
+      run: |
+        python3 -m pip list
+        python3 -m pip show xrootd
+        python3 -c 'import XRootD; print(XRootD)'
+        python3 -c 'import pyxrootd; print(pyxrootd)'
+        python3 -c 'from XRootD import client; print(client.FileSystem("root://someserver:1094"))'
+
+    - name: Run libXrdCl tests
+      run: |
+        cd ../build/tests
+        ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/UtilsTest/'
+        ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/SocketTest/'
+        ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/PollerTest/'
+        # ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/PostMasterTest/'
+        # ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/FileSystemTest/'
+        # ./common/test-runner ./XrdClTests/libXrdClTests.so 'All Tests/LocalFileHandlerTest/'
+
+  cmake-centos7-updated-python:
+
+    runs-on: ubuntu-latest
+    container: gitlab-registry.cern.ch/linuxsupport/cc7-base:latest
+
+    steps:
+    - name: Install external dependencies with yum
+      run: |
+        yum update -y
+        yum install --nogpg -y \
+            cmake3 \
+            make \
+            krb5-devel \
+            libuuid-devel \
+            libxml2-devel \
+            openssl-devel \
+            systemd-devel \
+            zlib-devel \
+            devtoolset-7-gcc-c++ \
+            which \
+            python3-devel \
+            python3-setuptools \
+            git \
+            cppunit-devel
+        yum clean all
+        python3 -m pip --no-cache-dir install --upgrade pip setuptools wheel
```

Comment:
I've added these `updated-python` jobs as it is important to check for changes that have happened in the `pip` and `setuptools` world that happen since CentOS 7 ships very old releases. Future Debian jobs will do a better job of checking this, as they have newer CPython releases while CentOS's Python 3.6 is capped out at [`setuptools` `59.6.0`](https://pypi.org/project/setuptools/59.6.0/) as [`setuptools` `59.7.0`+ is Python 3.7+](https://pypi.org/project/setuptools/59.7.0/).